src/Compilers/VisualBasic/Portable/Lowering/AsyncRewriter/AsyncRewriter.vb

```diff
@@ -292,15 +292,23 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                                          currentMethod:=currentMethod)
         End Function
 
+        ''' <returns>
+        ''' Returns true if any members that we need are missing or have use-site errors.
+        ''' </returns>
         Friend Overrides Function EnsureAllSymbolsAndSignature() As Boolean
-            Dim hasErrors As Boolean = MyBase.EnsureAllSymbolsAndSignature
+            If MyBase.EnsureAllSymbolsAndSignature Then
+                Return True
+            End If
```

Comment:
I agree using `bag` rather than `hasErrors` in ```Ensure*``` calls is cleaner. But to avoid returning early here, would it be sufficient to change the end of the method to:
```
If bag.HasAnyErrors Then
    hasErrors = True
End If
Diagnostics.AddRange(bag)
bag.Free()
Return hasErrors
```